### PR TITLE
[de] try to ignore words added as exception

### PR DESCRIPTION
**Pull request by Luisa-LT**
https://regression.languagetoolplus.com/via-http/2023-12-15/de-DE/result_java_OLD_SPELLING.html

Note: please do not merge (to merge after premium deploy)

### CODE REVIEW

File: languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/OldSpellingRule.java [hunk 1]
```diff
@@ -68,12 +78,11 @@ public RuleMatch[] match(AnalyzedSentence sentence) {
       }
       boolean ignore = false;
       for (String exception : EXCEPTIONS) {
-        if (hit.begin + exception.length() <= text.length()) {
-          String excCovered = text.substring(hit.begin, hit.begin + exception.length());
-          if (excCovered.equals(exception)) {
-            ignore = true;
-            break;
-          }
+        String pattern = "(?i)\\b" + Pattern.quote(exception) + "\\b";
```
Comment (danielnaber) [inline languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/OldSpellingRule.java:81]: This line/variable doesn't seem to be used

Comment (Luisa-LT) [reply languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/OldSpellingRule.java:81]: Should I undo the changes again?

Comment (danielnaber) [reply languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/OldSpellingRule.java:81]: Not needed, but you could delete this line, as it's not needed. IntelliJ should also show `pattern` in light gray, indicating the variable is never used.

Comment (Luisa-LT) [reply languagetool-language-modules/de/src/main/java/org/languagetool/rules/de/OldSpellingRule.java:81]: Thanks!
commit: https://github.com/languagetool-org/languagetool/pull/9967/commits/200fe554a26222f2224b2209e3d7dd9ea9e03000

**Pull request merged**